File: dayNine/python/test_mini_parking_functions.py
from unittest import TestCase

from mini_parking_functions import *

class TestMiniParkingFunctions(TestCase):

    def test_ThatWhen_iCheckTheParkingSpace_iGetAllThe_parkingSpaceEmpty(self):
        actualSlotAvailable = check_parking_space()
        expectedSlotAvailable = ["Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty"]
        
        self.assertEqual(expectedSlotAvailable,actualSlotAvailable)
        
    def test_thatWhen_iSelectedSlot_theSlotGetFilled_andThatPlace_andShowsParked(self):
        slot = 1;
        choose_slot(slot);
        
        actualSlotAvailable = check_parking_space()
        expectedSlotAvailable = ["Parked","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty"]
        
        self.assertEqual(expectedSlotAvailable,actualSlotAvailable)
        
    def test_thatWhen_iCheckoutParkingSlot_becomesEmpty(self):
        slot = 1
        choose_slot(slot)
        
        actualSlotAvailable = check_parking_space()
        expectedSlotAvailable = ["Parked","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty"]
        
        self.assertEqual(expectedSlotAvailable,actualSlotAvailable)
        
        slotTwo = 1
        check_out(slot)
        
        actualSlotNowAvailable = check_parking_space()
        expectedSlotNowAvailable = ["Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty"]
        
        self.assertEqual(expectedSlotNowAvailable,actualSlotNowAvailable)
        
    def test_thatWhen_iCheckASlot_itReturnsFalse_ifFilled(self):
        slot = 1
        choose_slot(slot)
        
        actualAnswer = is_slot_empty(slot)
        expectedAnswer = False
        
        self.assertEqual(expectedAnswer,actualAnswer)

File: dayNine/Java/TestMiniParkingFunctions.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class TestMiniParkingFunctions {

    @Test
    void TestThatWhenICheckTheParkingSpaceIGetAllTheParkingSpaceEmpty(){
        MiniParkingFunctions park = new MiniParkingFunctions();
        
        //When
        String [] actualSlotAvailable = park.checkParkingSpace();
        String [] expectedSlotAvailable = {"Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty"};
        
        //Check
        assertArrayEquals(expectedSlotAvailable,actualSlotAvailable);
    }
    
    @Test
    void TestThatWhenISelectedSlotSlotGetFilledAndThatPlaceAndShowsParked(){
        MiniParkingFunctions park = new MiniParkingFunctions();
        //Given
        int slot = 1;
        park.chooseSlot(slot);
        
        //When
        String [] actualSlotAvailable = park.checkParkingSpace();
        String [] expectedSlotAvailable = {"Parked","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty"};
        
        //Check
        assertArrayEquals(expectedSlotAvailable,actualSlotAvailable);
    }
    
    @Test
    void TestThatWhenICheckoutParkingSlotBecomesEmpty(){
        MiniParkingFunctions park = new MiniParkingFunctions();
        //Given
        int slot = 1;
        park.chooseSlot(slot);
        
        String [] actualSlotAvailable = park.checkParkingSpace();
        String [] expectedSlotAvailable = {"Parked","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty"};
        
        assertArrayEquals(expectedSlotAvailable,actualSlotAvailable);
        
        //Given
        int slotTwo = 1;
        park.checkOut(slot);
        
        //When
        String [] actualSlotNowAvailable = park.checkParkingSpace();
        String [] expectedSlotNowAvailable = {"Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty","Empty"};
        
        //Check
        assertArrayEquals(expectedSlotNowAvailable,actualSlotNowAvailable);
    }
    
    @Test
    void TestThatWhenICheckASlotItReturnsFalseIfFilled(){
        MiniParkingFunctions park = new MiniParkingFunctions();
        //Given
        int slot = 1;
        park.chooseSlot(slot);
        
        //When
        boolean actualAnswer = park.isSlotEmpty(slot);
        boolean expectedAnswer = false;
        
        //Check
        assertEquals(expectedAnswer,actualAnswer);
    }
}








